### PR TITLE
Add `options.query` to KMLDataSource

### LINKED ISSUE #5408: CZML and KML need the ability to pass query parameters
`CzmlDataSource` and `KmlDataSource` have no way to specify additional parameters that need to be included in url requests.  The most common use for this is query-based access tokens.

Both classes' `load` function (and in CZML's case the `process` function) should take an additional `query` object, off of the options object.  It would be a dictionary of strings that get converted to a query string via `objectToQuery` and globbed onto all urls requests.

### PR DESCRIPTION
**Pull request by omh1280**
Fixes #5408 

Pass around a `query` parameter which eventually gets blobbed onto the various uri in `KMLDataSource`. I've added like 6 unit tests, but am unconvinced about the thoroughness of this implementation. @mramato please test elsewhere? :)

One problem: using `proxyUrl` to append `query` means that `processNetworkLinkQueryString` will be using the already-formatted `href` ([LINE](https://github.com/AnalyticalGraphicsInc/cesium/compare/master...omh1280:kml_query?expand=1#diff-0aabbb637a10ce9c28e4d24c9f73a99eR1953)). Therefore, it will be used throughout the text-replacement process of `processNetworkLinkQueryString`. I don't think this will (ever) be a problem, unless some authentication token contains `[bboxWest]` or something.

I've also moved a [function call](https://github.com/AnalyticalGraphicsInc/cesium/compare/master...omh1280:kml_query?expand=1#diff-0aabbb637a10ce9c28e4d24c9f73a99eL492) out of an `if` block, so now the behavior is that `proxyUrl` is run for every URI. This is ok because if either `proxy` or `query` are `undefined`, `proxyUrl` doesn't do anything.

Ottavio

### CODE REVIEW

File: Source/DataSources/KmlDataSource.js [hunk 1]
```diff
@@ -489,8 +494,8 @@ define([
         }
         if (!hrefResolved && defined(sourceUri)) {
             href = getAbsoluteUri(href, getAbsoluteUri(sourceUri));
-            href = proxyUrl(href, proxy);
         }
+        href = proxyUrl(href, proxy, query);
```
Comment (mramato) [inline Source/DataSources/KmlDataSource.js]: This still needs to be inside of an `if (!hrefResolved)` block, so adjust the original logic to handle that.  Basically, if `hrefResolved` is true at this point, it means that the item is a data or blob uri, which means we can't/shouldn't append query parameters to it.

**Comment by mramato** [comment]
At first glance, this looks perfect and that was the only comment I had.  Also update CHANGES (just edit the CZML line from your last PR).

I'll put this through its paces and take a deeper look and let you know if I find any problems.

Thanks @omh1280!

**Comment by mramato** [comment]
`processBillboardIcon` needs to be passed a copy of the query (which means it needs to be threaded through `apply style` as well.  (it also means you have at least one missing test to cover that code.)

File: Specs/DataSources/KmlDataSourceSpec.js [hunk 2]
```diff
@@ -1555,6 +1555,32 @@ defineSuite([
...
+          <Placemark>\
+              <Style>\
+                  <IconStyle>\
+                      <Icon>\
+                          <href>image.png</href>\
+                      </Icon>\
+                  </IconStyle>\
+              </Style>\
+          </Placemark>';
+        debugger;
```
Comment (mramato) [inline Specs/DataSources/KmlDataSourceSpec.js]: You forgot to take this out (always try and remember to `npm run eslint` before each commit/push)

Comment (omh1280) [inline Specs/DataSources/KmlDataSourceSpec.js]: Fixed, thanks

**Comment by mramato** [comment]
Awesome work here @omh1280!  Thanks.

**Pull request merged**